spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteViewCommands.scala

```diff
@@ -60,17 +62,18 @@ case class RewriteViewCommands(spark: SparkSession) extends Rule[LogicalPlan] wi
         properties = properties,
         allowExisting = allowExisting,
         replace = replace)
+
+    case u@UnresolvedView(ResolvedView(resolved), _, _, _) =>
```

Comment:
Why does this need to be done here instead of in `ResolveViews`? This is just resolving the view. I think it's probably that we have to resolve early to avoid an `AnalysisException`, but it would be great to clarify with a comment.

Reply:
yes this needs to be done early because otherwise the Analyzer will resolve views and fail (since v2 views aren't supported). I'll add a comment